src/Generator/ValueGenerator.php

```diff
@@ -143,7 +149,7 @@ public function deleteConstant($constant)
     /**
      * Return constant list
      *
-     * @return ArrayObject
+     * @return SplArrayObject|StdlibArrayObject
```

Comment:
ARGH!!!! I stand corrected: `Zend\Stdlib\ArrayObject` does **not** extend `ArrayObject`. I forgot that we implemented our own version due to limitations in the PHP implementation. It has the same *behavior*, but a different *inheritance* chain.

As such, the approach in this PR is a bit suspect. Type-hinting on `ArrayAccess` *requires* an additional check later for `ArrayObject` or `Zend\Stdlib\ArrayObject`, because `ValueGenerator` consumes the *methods* `getArrayCopy()`,  `exchangeArray()`, and `append()` from those two implementations (in addition to `offsetUnset()` from `ArrayAccess`, and supporting iteration).

I'd argue:

- No typehint for the argument (drop the `ArrayAccess` typehint, as it incorrectly indicates any `ArrayAccess` implementation will work).
- In the docblock, specify the two implementations as the typehint (`\ArrayObject|\Zend\Stdlib\ArrayObject` — as it is already)
- Check in the constructor for an instance of one of those two implementations (as is already being done). 

With those changes, I'll gladly merge.